ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx

```diff
@@ -113,30 +48,36 @@ export class Sharing extends React.Component<Props, State> {
       items: [],
     }
   }
-
   componentDidMount = () => {
     fetch(`/search/catalog/internal/metacard/${this.props.id}`)
       .then(response => response.json())
       .then(data => {
         const metacard = data.metacards[0]
-        const items = getGroups(
-          metacard[Security.GroupsWrite] || [],
-          metacard[Security.GroupsRead] || []
-        ).concat(
-          getIndividuals(
-            metacard['metacard.owner'] || [],
-            metacard[Security.IndividualsWrite] || [],
-            metacard[Security.IndividualsRead] || [],
-            metacard[Security.AccessAdministrators] || []
-          )
-        )
-        this.setState({ items: items })
+        const res = Restrictions.from(metacard)
+        const security = new Security(res)
+        const individuals = security.getIndividuals().map((e: Entry) => {
+          return {
+            ...e,
+            id: common.generateUUID(),
+            category: Category.User,
+            visible: e.value !== res.owner, // hide owner
+          } as Item
+        })
+        const groups = security.getGroups(user.getRoles()).map((e: Entry) => {
+          return {
+            ...e,
+            id: common.generateUUID(),
+            category: Category.Group,
+            visible: user.getRoles().indexOf(e.value) > -1, // only display the groups the current user has
+          } as Item
+        })
+        this.setState({ items: groups.concat(individuals) })
         this.add()
```

Comment:
❓ I know you didn't touch this, but what this method is doing doesn't smell right. Direct state modifications do not trigger a re-render. Is this intentional?

Reply:
are we talking about the this.add()?

Reply:
this.add() calls this.setState() at the end

Reply:
https://github.com/codice/ddf/blob/9379e9ebdd53e323c31ed0c6fd50eaa0c56bef18/ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx#L163